CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc

```diff
@@ -450,37 +451,34 @@ std::unique_ptr<HcalRespCorrs> HcalHardcodeCalibrations::produceRespCorrs (const
  
   auto result = std::make_unique<HcalRespCorrs>(topo);
   std::vector <HcalGenericDetId> cells = allCells(*topo, dbHardcode.killHE());
-  for (std::vector <HcalGenericDetId>::const_iterator cell = cells.begin (); cell != cells.end (); ++cell) {
+  for (const auto& cell : cells) {
 
     double corr = 1.0; 
 
-    if ((he_recalibration != 0 ) && 
-	((*cell).genericSubdet() == HcalGenericDetId::HcalGenEndcap)) {
-      
-      int depth_ = HcalDetId(*cell).depth();
-      int ieta_  = HcalDetId(*cell).ieta();
-      corr = he_recalibration->getCorr(ieta_, depth_); 
-      
-      /*
-	std::cout << "HE ieta, depth = " << ieta_  << ",  " << depth_  
-	<< "   corr = "  << corr << std::endl;
-      */
+    //check for layer 0 reweighting: when depth 1 has only one layer, it is layer 0
+    if( useLayer0Weight && 
+      ((cell.genericSubdet() == HcalGenericDetId::HcalGenEndcap) || (cell.genericSubdet() == HcalGenericDetId::HcalGenBarrel)) &&
+      (HcalDetId(cell).depth()==1 && dbHardcode.getLayersInDepth(HcalDetId(cell).ieta(),HcalDetId(cell).depth(),topo)==1) )
+    {
+      corr = 0.5/1.2;
+    }
 
+    if ((he_recalibration != 0 ) && (cell.genericSubdet() == HcalGenericDetId::HcalGenEndcap)) {
+      int depth_ = HcalDetId(cell).depth();
+      int ieta_  = HcalDetId(cell).ieta();
+      corr *= he_recalibration->getCorr(ieta_, depth_); 
+      
+      //std::cout << "HE ieta, depth = " << ieta_  << ",  " << depth_ << "   corr = "  << corr << std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant